src/common/libutil/strstrip.c

```diff
@@ -0,0 +1,78 @@
+/************************************************************\
+ * Copyright 2022 Lawrence Livermore National Security, LLC
+ * (c.f. AUTHORS, NOTICE.LLNS, COPYING)
+ *
+ * This file is part of the Flux resource manager framework.
+ * For details, see https://github.com/flux-framework.
+ *
+ * SPDX-License-Identifier: LGPL-3.0
+\************************************************************/
+
+#if HAVE_CONFIG_H
+#include "config.h"
+#endif
+
+#include <string.h>
+#include <errno.h>
+#include <ctype.h>
+
+char *strstrip (char *s)
+{
+    size_t size;
+    char *end;
+
+    if (!s) {
+        errno = EINVAL;
+        return NULL;
+    }
+
+    size = strlen (s);
+
+    if (!size)
+        return s;
+
+    end = s + size - 1;
+    while (end >= s && isspace ((unsigned char) *end))
+        end--;
+    *(end + 1) = '\0';
+
+    while (*s && isspace((unsigned char) *s))
+        s++;
+
+    return s;
+}
+
+char *strstrip_copy (const char *s)
+{
+    size_t size;
+    char *end;
+    char *result = NULL;
+
+    if (!s) {
+        errno = EINVAL;
+        return NULL;
+    }
+
+    if (s[0] == '\0')
+        return strdup (s);
+
+    /* Work from front to back so we do not have to copy all of 's'
+     */
+    while (*s && isspace ((unsigned char) *s))
+        s++;
+
+    if (!(result = strdup (s)))
+        return NULL;
+
+    size = strlen (result);
+
+    end = result + size - 1;
+    while (end >= result && isspace ((unsigned char) *end))
+        end--;
+    *(end + 1) = '\0';
+
+    return result;
```

Comment:
i think this is safe, but size could be 0 after the front striping of spaces (i.e. string of all spaces), so we're sort of doing pointer arithmetic with an invalid pointer.  Perhaps add a ` if (s[0] == '\0')` check?

Reply:
Yeah, if `size == 0` then `end = result - 1` so `*(end + 1) = '\0'` just reterminates `result`. I see what you mean, we might as well do this:
```c
    if ((size = strlen (result)) == 0)
        return result;
```